test/e2e/storage/volumeattributesclass.go

```diff
@@ -171,4 +184,141 @@ var _ = utils.SIGDescribe("VolumeAttributesClass", feature.VolumeAttributesClass
 		}))
 		framework.ExpectNoError(err, "timeout while waiting to confirm VolumeAttributesClass %q deletion", updatedVolumeAttributesClass.Name)
 	})
+
+	framework.It("should be protected by vac-protection finalizer", func(ctx context.Context) {
+		vacClient := f.ClientSet.StorageV1beta1().VolumeAttributesClasses()
+
+		initialVAC := &storagev1beta1.VolumeAttributesClass{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "VolumeAttributesClass",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+			DriverName: "e2e-fake-csi-driver",
+			Parameters: map[string]string{
+				"foo": "bar",
+			},
+		}
+
+		ginkgo.By("Creating a VolumeAttributesClass")
+		createdVAC, err := vacClient.Create(ctx, initialVAC, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Failed to create the requested VolumeAttributesClass")
+		ginkgo.DeferCleanup(framework.IgnoreNotFound(vacClient.Delete), createdVAC.Name, metav1.DeleteOptions{})
+
+		ginkgo.By("Verifying that the vac-protection finalizer is set")
+		err = waitForVolumeAttributesClassFinalizerSet(ctx, f.ClientSet, createdVAC.Name, framework.Poll, VacUpdatingTimeout)
+		framework.ExpectNoError(err)
+
+		// Simulate VAC being in use
+		ginkgo.By("Creating a PVC that uses the VolumeAttributesClass")
+		pvc := &v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-pvc-",
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				VolumeAttributesClassName: &createdVAC.Name,
+				AccessModes:               []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+				Resources:                 v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("1")}},
+			},
+		}
```

Comment:
Please set `spec.storageClassName` to `""`. It avoids creating a PV if the cluster has a default sc. And adds a selector like `"x-test.k8s.io/unsatisfiable": "not-bindable"` to avoid bind an existing PV.